tests/test_accumulators.py

```diff
@@ -85,3 +87,47 @@ def test_mean():
     assert a == bh.accumulators.Mean(3, 2, 1)
 
     assert repr(a) == "Mean(count=3, value=2, variance=1)"
+
+
+def test_sum_mean():
+    a = bh.accumulators.Mean()
+    a.fill([1, 2, 3])
+
+    b = bh.accumulators.Mean()
+    b.fill([5, 6])
+
+    c = bh.accumulators.Mean()
+    c.fill([1, 2, 3, 5, 6])
+
+    ab = a + b
+    assert ab.value == approx(c.value)
+    assert ab.variance == approx(c.variance)
+    assert ab.count == approx(c.count)
+
+    a += b
+    assert a.value == approx(c.value)
+    assert a.variance == approx(c.variance)
+    assert a.count == approx(c.count)
+
+
+def test_sum_weighed_mean():
+    a = bh.accumulators.WeightedMean()
+    a.fill([1, 2, 3])
+
+    b = bh.accumulators.WeightedMean()
+    b.fill([5, 6])
+
+    c = bh.accumulators.WeightedMean()
+    c.fill([1, 2, 3, 5, 6])
```

Comment:
This test is incomplete if you don't use weights.

Reply:
To clarify, if you don't use the original Boost Histogram code, then you should fully test the implementation thoroughly. If you rely on the original Boost Histogram code, then you can be more lax and just test the interface.

Reply:
Added weights, and added a test for the histogram summing as seen in the issue this fixes. As mentioned elsewhere, we have to have a copy because Boost.Histogram makes the internal memory storage private, but we have to be able to export this to a NumPy dtype, and the Python code has to internally convert back and forth, such as making an accumulator from a View, pickling, etc. You helped me do this in #194.